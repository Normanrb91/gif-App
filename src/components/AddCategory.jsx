import { useForm } from '../hooks/useForm'

export const AddCategory = ({ onNewCategory }) => {
  const [values, handleInputChange, reset] = useForm({ category: '' })

  const handleSubmit = e => {
    e.preventDefault()
    if (values.category.trim().length <= 1) return

    onNewCategory(values.category)
    reset()
  }

  return (
    <form onSubmit={handleSubmit}>
      <input
        placeholder="Buscar Gifs"
        name="category"
        type="text"
        value={values.category}
        onChange={handleInputChange}
      />
    </form>
  )
}
