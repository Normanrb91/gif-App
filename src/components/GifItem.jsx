export const GifItem = ({ gif }) => {
  return <div>{gif.title}</div>
}
