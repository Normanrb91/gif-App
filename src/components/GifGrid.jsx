import { GifItem } from './GifItem'
import { useFetchGifs } from '../hooks/useFetchGifs'

export const GifGrid = ({ category }) => {
  const { data, loading } = useFetchGifs(category)

  return (
    <>
      <h3>{category}</h3>

      {loading ? (
        <h2>Cargando...</h2>
      ) : (
        <div className="card-grid">
          {data.map(gif => (
            <GifItem key={gif.id} {...gif} />
          ))}
        </div>
      )}
    </>
  )
}
