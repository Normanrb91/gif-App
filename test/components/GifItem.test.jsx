import { render, screen } from '@testing-library/react'
import { GifItem } from '../../src/components/GifItem'

describe('pruebas en <GifItem />', () => {
  const title = 'betis'
  const url = 'https://betis.com/plantilla.jpg'

  test('debe hacer match con snapshot', () => {
    const { container } = render(<GifItem title={title} url={url} />)
    expect(container).toMatchSnapshot()
  })

  test('debe de mostrar la imagen con la URL y ALT indicado', () => {
    render(<GifItem title={title} url={url} />)

    const { src, alt } = screen.getByRole('img')
    expect(src).toBe(url)
    expect(alt).toBe(title)
  })

  test('debe de mostrar el titulo en el componente', () => {
    render(<GifItem title={title} url={url} />)
    expect(screen.getByText(title)).toBeTruthy()
  })
})
