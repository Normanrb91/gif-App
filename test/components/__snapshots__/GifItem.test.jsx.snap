// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifItem /> debe hacer match con snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="betis"
      src="https://betis.com/plantilla.jpg"
    />
    <p>
      betis
    </p>
  </div>
</div>
`;
